iterating-over-data/iterating-over-dictionaries.md

```diff
@@ -0,0 +1,321 @@
+# Iterating Over Dictionaries
+
+## Learning Goals
+
+- Practice using for loops to iterate through a dictionary
+
+## Introduction
+
+The overall concept for iterating over dictionaries is the same as iterating over lists. We can use `for` loops to iterate over **each key-value pair in a dictionary.**
+
+However, because dictionaries are key-value pairs, there are a few differences to pay attention to.
+
+## `for` Loops for Keys and Values Need `.items()`
+
+To iterate over a dictionary, we can use the following syntax:
+
+```python
+for my_key, my_value in my_dict.items():
+    print(my_key, my_value)
+```
+
+| Piece of Code              | Notes                                                                                                                                                        |
+| -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `for`                      | `for` is a reserved keyword in Python. Python recognizes `for` as the beginning of a `for` loop.                                                             |
+| `my_key`                   | **Replace this** part with a name that represents what each key is. This local variable represents the _key_ in each key-value pair, one at a time.          |
```

Comment:
```suggestion
| `my_key`                   | **Replace this** part with a name that represents what each key is. During iteration, this will be a variable that will receive the _key_ in each key-value pair, one at a time.          |
```

Reply:
My concern about this phrasing (letting During iteration decorate the whole phrase) is that it's unclear whether the variable is only accessible during the iteration (it isn't).  Maybe this???
```suggestion
| `my_key`                   | **Replace this** part with a name that represents what each key is. This will be used as a variable name that will receive the _key_ in each key-value pair during the iteration, one key at a time.          |
```